### PR TITLE
Feature/template.swift templateizable through cli

### PR DESCRIPTION
**Pull request by xmoba-pablo**
Now, you can indicate the template you want use it. 

You can customize the template!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,8 +73,9 @@ CLI usage:
 
 ```
 $ ./unstringify
-Usage: ./unstringify inputPath outputPath
+Usage: ./unstringify inputPath outputPath templatePath
```
Comment (hectr) [inline README.md]: I would put templatePath in brackets, to make it clear it is optional:
`Usage: ./unstringify inputPath outputPath [templatePath]`

File: TemplateExample.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Generated by Unstringify.
```
Comment (hectr) [inline TemplateExample.swift]: I would move this file to the Examples directory.

Also I would change this line to:
`// Generated by Unstringify with a custom template`

File: TemplateExample.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        let bundleURL = _UnstringifiedBundle.bundleURL
+        let bundleName = bundleURL.lastPathComponent
+        let resource = (bundleName as NSString).deletingPathExtension
+        guard let path = _UnstringifiedBundle.path(forResource: resource, ofType: "bundle") else {
+            return nil
+        }
+        return Bundle(path: path)
+    }
+}
+
```
Comment (hectr) [inline TemplateExample.swift]: In this custom template example I would replace the following lines with:
```
public enum Localizable {
     public enum Text: String, Unstringified {
          public typealias StringType = String
          case $KEYS_ARRAY
      }

       public enum Format: Unstringified {
          public typealias StringType = String
          case $FORMATED_KEYS_ARRAY
      }

       public enum RichText: String, Unstringified {
          public typealias StringType = NSAttributedString
          case $RICH_KEYS_ARRAY
      }

       public enum RichFormat: Unstringified {
          public typealias StringType = NSAttributedString
          case $FORMATED_RICH_KEYS_ARRAY
      }
}
```

File: Sources/unstringify/Internal/template.swift [hunk 4]
```diff
@@ -1,22 +1,33 @@
...
-    let file: String =
+func template(keys: [String], formattedKeys: [FormatKey], richKeys: [String], richFormattedKeys: [FormatKey], templatePath: String? = nil) -> String {
+    do {
+        if let templatePath = templatePath {
+            var templateContent = try readFile(at: templatePath)
+            templateContent = replaceVariables(content: templateContent, keys: keys, formattedKeys: formattedKeys, richKeys: richKeys, richFormattedKeys: richFormattedKeys)
+            return templateContent
+        }
+    } catch {
+
```
Comment (CSolanaM) [inline Sources/unstringify/Internal/template.swift]: Unhandled error fallback in catch clause. Should fail gracefully when readFile fails.

**Review by CSolanaM — APPROVED** [review]
👍

**Review by hectr — APPROVED** [review]
👍 Good job @xmoba-pablo !

File: Sources/unstringify/Internal/template.swift (hunk 4 above)
Comment (xmoba-pablo) [reply Sources/unstringify/Internal/template.swift]: Done!

File: README.md (hunk 1 above)
Comment (xmoba-pablo) [reply README.md]: Done!

File: TemplateExample.swift (hunk 2 above)
Comment (xmoba-pablo) [reply TemplateExample.swift]: Done!

File: TemplateExample.swift (hunk 3 above)
Comment (xmoba-pablo) [reply TemplateExample.swift]: Done!

File: Sources/unstringify/Internal/template.swift (hunk 4 above)
Comment (hectr) [reply Sources/unstringify/Internal/template.swift]: I disagree, just don't catch this error if you cannot recover from it

Comment (hectr) [reply Sources/unstringify/Internal/template.swift]: for that you'll need to add `throws` to the method signature

**Pull request merged**